choreolib/src/main/java/choreo/autos/AutoFactory.java

```diff
@@ -173,6 +173,9 @@ public <SampleType extends TrajectorySample<SampleType>> AutoFactory(
    * @see #voidLoop
    */
   public AutoLoop newLoop(String name) {
+    // causing a cache clear in simulation to allow a form of `hot-reloading`
+    // for trajectories in simulation
+    if (RobotBase.isSimulation()) clearCache();
```

Comment:
This change should be ported to C++. We've also generally avoided single-line if statements in the rest of the codebase in favor of braces.